{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "from keras.models import load_model\n",
    "import cv2\n",
    "import numpy as np\n",
    "from playsound import playsound"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: playsound in c:\\users\\asus\\anaconda3\\lib\\site-packages (1.3.0)\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip install playsound"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "model = load_model('model-002.model')\n",
    "\n",
    "face_clsfr=cv2.CascadeClassifier('haarcascade_frontalface_default.xml')\n",
    "\n",
    "source=cv2.VideoCapture(0)\n",
    "\n",
    "labels_dict={1:'MASK',0:'NO MASK'}\n",
    "color_dict={1:(0,255,0),0:(0,0,255)}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "    Error 259 for command:\n",
      "        play \"C:\\Users\\ASUS\\AppData\\Local\\Temp\\PSg31x4iy5.mp3\" wait\n",
      "    The driver cannot recognize the specified command parameter.\n",
      "\n",
      "    Error 259 for command:\n",
      "        play \"C:\\Users\\ASUS\\AppData\\Local\\Temp\\PS28nxioeu.mp3\" wait\n",
      "    The driver cannot recognize the specified command parameter.\n",
      "\n",
      "    Error 305 for command:\n",
      "        close \"C:\\Users\\ASUS\\AppData\\Local\\Temp\\PSg31x4iy5.mp3\"\n",
      "    Cannot specify extra characters after a string enclosed in quotation marks.\n",
      "\n",
      "    Error 305 for command:\n",
      "        close \"C:\\Users\\ASUS\\AppData\\Local\\Temp\\PS28nxioeu.mp3\"\n",
      "    Cannot specify extra characters after a string enclosed in quotation marks.\n",
      "Failed to close the file: \"C:\\Users\\ASUS\\AppData\\Local\\Temp\\PSg31x4iy5.mp3\"\n",
      "Exception in thread Failed to close the file: \"C:\\Users\\ASUS\\AppData\\Local\\Temp\\PS28nxioeu.mp3\"\n",
      "Thread-91:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "Exception in thread Thread-93:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\threading.py\", line 910, in run\n",
      "    self.run()\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\site-packages\\playsound.py\", line 44, in _playsoundWin\n",
      "    _playsoundWin(tempPath, block)\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\site-packages\\playsound.py\", line 73, in _playsoundWin\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\site-packages\\playsound.py\", line 44, in _playsoundWin\n",
      "    winCommand(u'play {}{}'.format(sound, ' wait' if block else ''))\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\site-packages\\playsound.py\", line 64, in winCommand\n",
      "    _playsoundWin(tempPath, block)\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\site-packages\\playsound.py\", line 73, in _playsoundWin\n",
      "    raise PlaysoundException(exceptionMessage)\n",
      "playsound.PlaysoundException: \n",
      "    Error 259 for command:\n",
      "        play \"C:\\Users\\ASUS\\AppData\\Local\\Temp\\PSg31x4iy5.mp3\" wait\n",
      "    The driver cannot recognize the specified command parameter.\n",
      "    winCommand(u'play {}{}'.format(sound, ' wait' if block else ''))\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\site-packages\\playsound.py\", line 64, in winCommand\n",
      "    raise PlaysoundException(exceptionMessage)\n",
      "playsound.PlaysoundException: \n",
      "    Error 259 for command:\n",
      "        play \"C:\\Users\\ASUS\\AppData\\Local\\Temp\\PS28nxioeu.mp3\" wait\n",
      "    The driver cannot recognize the specified command parameter.\n",
      "\n",
      "    Error 259 for command:\n",
      "        play \"C:\\Users\\ASUS\\AppData\\Local\\Temp\\PSxa1cj0xn.mp3\" wait\n",
      "    The driver cannot recognize the specified command parameter.\n",
      "\n",
      "    Error 259 for command:\n",
      "        play \"C:\\Users\\ASUS\\AppData\\Local\\Temp\\PSdqpq3x83.mp3\" wait\n",
      "    The driver cannot recognize the specified command parameter.\n",
      "\n",
      "    Error 259 for command:\n",
      "        play \"C:\\Users\\ASUS\\AppData\\Local\\Temp\\PS1b1mzp2w.mp3\" wait\n",
      "    The driver cannot recognize the specified command parameter.\n",
      "\n",
      "    Error 259 for command:\n",
      "        play \"C:\\Users\\ASUS\\AppData\\Local\\Temp\\PSw7stxhry.mp3\" wait\n",
      "    The driver cannot recognize the specified command parameter.\n",
      "\n",
      "    Error 305 for command:\n",
      "        close \"C:\\Users\\ASUS\\AppData\\Local\\Temp\\PSxa1cj0xn.mp3\"\n",
      "    Cannot specify extra characters after a string enclosed in quotation marks.\n",
      "\n",
      "    Error 305 for command:\n",
      "        close \"C:\\Users\\ASUS\\AppData\\Local\\Temp\\PSdqpq3x83.mp3\"\n",
      "    Cannot specify extra characters after a string enclosed in quotation marks.\n",
      "\n",
      "    Error 305 for command:\n",
      "        close \"C:\\Users\\ASUS\\AppData\\Local\\Temp\\PS1b1mzp2w.mp3\"\n",
      "    Cannot specify extra characters after a string enclosed in quotation marks.\n",
      "Failed to close the file: \"C:\\Users\\ASUS\\AppData\\Local\\Temp\\PSxa1cj0xn.mp3\"\n",
      "Exception in thread Failed to close the file: \"C:\\Users\\ASUS\\AppData\\Local\\Temp\\PSdqpq3x83.mp3\"\n",
      "Thread-103:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "Exception in thread Thread-96:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "Failed to close the file: \"C:\\Users\\ASUS\\AppData\\Local\\Temp\\PS1b1mzp2w.mp3\"\n",
      "Exception in thread Thread-100:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\threading.py\", line 910, in run\n",
      "Exception in thread Thread-98:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\threading.py\", line 910, in run\n",
      "    self.run()    \n",
      "self.run()\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\threading.py\", line 910, in run\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\site-packages\\playsound.py\", line 44, in _playsoundWin\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\site-packages\\playsound.py\", line 44, in _playsoundWin\n",
      "        self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\site-packages\\playsound.py\", line 44, in _playsoundWin\n",
      "_playsoundWin(tempPath, block)\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\site-packages\\playsound.py\", line 73, in _playsoundWin\n",
      "    _playsoundWin(tempPath, block)\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\site-packages\\playsound.py\", line 73, in _playsoundWin\n",
      "        self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\site-packages\\playsound.py\", line 44, in _playsoundWin\n",
      "_playsoundWin(tempPath, block)\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\site-packages\\playsound.py\", line 73, in _playsoundWin\n",
      "    winCommand(u'play {}{}'.format(sound, ' wait' if block else ''))\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\site-packages\\playsound.py\", line 64, in winCommand\n",
      "    winCommand(u'play {}{}'.format(sound, ' wait' if block else ''))\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\site-packages\\playsound.py\", line 64, in winCommand\n",
      "    _playsoundWin(tempPath, block)\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\site-packages\\playsound.py\", line 73, in _playsoundWin\n",
      "    winCommand(u'play {}{}'.format(sound, ' wait' if block else ''))\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\site-packages\\playsound.py\", line 64, in winCommand\n",
      "    winCommand(u'play {}{}'.format(sound, ' wait' if block else ''))\n",
      "  File \"C:\\Users\\ASUS\\anaconda3\\lib\\site-packages\\playsound.py\", line 64, in winCommand\n",
      "    raise PlaysoundException(exceptionMessage)\n",
      "playsound.PlaysoundException: \n",
      "    Error 259 for command:\n",
      "        play \"C:\\Users\\ASUS\\AppData\\Local\\Temp\\PSw7stxhry.mp3\" wait\n",
      "    The driver cannot recognize the specified command parameter.\n",
      "    raise PlaysoundException(exceptionMessage)\n",
      "playsound.PlaysoundException: \n",
      "    Error 259 for command:\n",
      "        play \"C:\\Users\\ASUS\\AppData\\Local\\Temp\\PSdqpq3x83.mp3\" wait\n",
      "    The driver cannot recognize the specified command parameter.\n",
      "    raise PlaysoundException(exceptionMessage)\n",
      "playsound.PlaysoundException: \n",
      "    Error 259 for command:\n",
      "        play \"C:\\Users\\ASUS\\AppData\\Local\\Temp\\PSxa1cj0xn.mp3\" wait\n",
      "    The driver cannot recognize the specified command parameter.\n",
      "    raise PlaysoundException(exceptionMessage)\n",
      "playsound.PlaysoundException: \n",
      "    Error 259 for command:\n",
      "        play \"C:\\Users\\ASUS\\AppData\\Local\\Temp\\PS1b1mzp2w.mp3\" wait\n",
      "    The driver cannot recognize the specified command parameter.\n"
     ]
    }
   ],
   "source": [
    "import threading\n",
    "\n",
    "while(True):\n",
    "\n",
    "    ret,img=source.read()\n",
    "    gray=cv2.cvtColor(img,cv2.COLOR_BGR2GRAY)\n",
    "    faces=face_clsfr.detectMultiScale(gray,1.3,5)  \n",
    "\n",
    "    for (x,y,w,h) in faces:\n",
    "    \n",
    "        face_img=gray[y:y+w,x:x+w]\n",
    "        \n",
    "        resized=cv2.resize(face_img,(100,100))\n",
    "        normalized=resized/255.0\n",
    "        reshaped=np.reshape(normalized,(1,100,100,1))\n",
    "        result=model.predict(reshaped)\n",
    "\n",
    "        label=np.argmax(result,axis=1)[0]\n",
    "        if(label==0):\n",
    "            threading.Thread(target = playsound, args = ('C:\\\\Users\\\\ASUS\\\\Documents\\\\PC\\\\College\\\\Events\\\\DROID 2.0\\\\Mask-detection\\\\beep.mp3', ), daemon = True).start()\n",
    "            \n",
    "      \n",
    "        cv2.rectangle(img,(x,y),(x+w,y+h),color_dict[label],2)\n",
    "        cv2.rectangle(img,(x,y-40),(x+w,y),color_dict[label],-1)\n",
    "        cv2.putText(img, labels_dict[label], (x, y-10),cv2.FONT_HERSHEY_SIMPLEX,0.8,(255,255,255),2)\n",
    "        \n",
    "        \n",
    "    cv2.imshow('FAce detection',img)\n",
    "    key=cv2.waitKey(1)\n",
    "    \n",
    "    if(key==27):\n",
    "        break\n",
    "        \n",
    "cv2.destroyAllWindows()\n",
    "source.release()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
